src/metabase/query_processor/pivot.clj

```diff
@@ -0,0 +1,20 @@
+(ns metabase.query-processor.pivot
+  "Pivot table actions for the query processor"
+  (:require [metabase.util.i18n :refer [trs]]))
+
+(defn- generate-breakouts
+  "Generate the combinatorial breakouts for a given query pivot table query"
+  [breakouts]
+  (when (not= 3 (count breakouts))
+    (throw (ex-info (trs "A pivot table query requires three breakouts") {:breakout breakouts})))
+
+  [(take 1 breakouts) (take 2 breakouts) (take-last 2 breakouts) []])
```

Comment:
This would be nicer/more generically expressed as a powerset problem (ie. generate all subsets).